### ISSUE TITLE
Prediction in real time using Keras

### ISSUE THREAD

**Issue by vasilevskykv** [tensorflow/models#8811]
Hi! I would like to ask, does Keras Neural Networks do predictions in real time?
I see, that one of your parameters in prediction is array of data. But when you need to predict future data, you don't have an array of data - you have only previous data and time interval for prediction and nothing more.
As I undestand, you simply checked the prediction because to predict future data I need to give to a function predict a time interval in future, not a previous data trainX as necessary in this function predict

Here is my Neuro Network:
look_back = 3
model = Sequential()
model.add(LSTM(4, input_shape=(look_back, 1)))
model.add(Dense(1))
model.compile(loss='mean_squared_error', optimizer='adam')
model.fit(trainX, trainY, epochs=20, batch_size=1, verbose=2)
trainPredict = model.predict(trainX)

Can you show an example of predicting in real time?

**Comment by ravikyram** [comment]
@vasilevskykv 

Please, fill [issue template.](https://github.com/tensorflow/models/issues/new/choose).

Please, let us know which pretrained model you are using and share related code .Thanks!

**Comment by vasilevskykv** [comment]
I would like to predict future values of EMA oscillator using previous values and a value in the current time moment
data2 = pandas.read_csv('ratevalues.csv')[::-1]
data2.columns = ['date','high','low','open','close','volume','quoteVolume','weightedAverage']
dataset = pandas.DataFrame(data2.close.ewm(span=14).mean())
train_size = int(len(dataset) * 0.67)
train = dataset[0:train_size,:]
trainX, trainY = create_dataset(train, look_back)

**Comment by vasilevskykv** [comment]
def create_dataset(dataset, look_back=1):
	dataX, dataY = [], []
	for i in range(len(dataset)-look_back-1):
		a = dataset[i:(i+look_back), 0]
		dataX.append(a)
		dataY.append(dataset[i + look_back, 0])
	return np.array(dataX), np.array(dataY)

**Comment by ravikyram** [comment]
@vasilevskykv 

if you are not using any predefined research or official models, please raise this issue in tf core repo from [here](https://github.com/tensorflow/tensorflow/issues/new/choose).Thanks!

**Comment by vasilevskykv** [comment]
I made a new issue here:
https://github.com/tensorflow/tensorflow/issues/41274

**Comment by ravikyram** [comment]
@vasilevskykv 

Can we close the issue here and track the issue in ` tensorflow/tensorflow#41274`.Thanks!

**Comment by vasilevskykv** [comment]
Yes, we can close
